cmake/add_enclave.cmake

```diff
@@ -1,50 +1,96 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 #
-# Helper function to sign an enclave binary.
+# Helper mecro to either sign an SGX enclave binary or to generate and sign
+# an OP-TEE-compatible enclave binary.
 #
 # Usage:
 #
 #  add_enclave(<TARGET target>
+#              [<UUID uuid>]
 #              [CXX]
 #              <SOURCES sources>
 #              [<CONFIG config>]
 #              [<KEY key>])
 #
+# For SGX enclaves:
 # Given <target> and <config>, this function adds custom commands to
 # generate a signing key if key is not specified and call `oesign` to
 # sign the target, resulting in `<target>.signed`. It also adds
 # `<target>_signed` as an imported target so that it can be referenced
 # later in the CMake graph.
 #
+# For OP-TEE enclaves:
+# Given <target> and <uuid>, this function adds custom commands to
+# generate and sign an OP-TEE Trusted Application (TA), the equivalent
+# of an enclave for ARM TrustZone. TA binaries must follow a specific
+# layout to be loadable by OP-TEE's loader. This macro helps ensure that
+# that layout is indeed followed, adding a `<target>.ta` target to generate
+# the final TA binary. Additionally, TA binaries must be named with a UUID.
+# As such, while the CMake target names remain in sync with <target>, the
+# resulting binaries use <uuid> plus their corresponding extension as their
+# on-disk name.
+#
 # The target is always linked to `oeenclave`, and if the optional flag
 # `CXX` is passed, it is also linked to `oelibcxx`
 #
+# NOTE: This must be a macro! To generate TA binaries, a custom linker
```

Comment:
Thanks for documenting and explaining this.

Reply:
No problem!